runtime/moonbase/src/lib.rs

```diff
@@ -384,6 +399,31 @@ where
 	}
 }
 
+pub struct BlockAuthorAccountId<R>(PhantomData<R>);
+impl<R> TypedGet for BlockAuthorAccountId<R>
+where
+	R: frame_system::Config + pallet_author_inherent::Config,
+	pallet_author_inherent::Pallet<R>: Get<R::AccountId>,
+{
+	type Type = R::AccountId;
+	fn get() -> Self::Type {
+		<pallet_author_inherent::Pallet<R> as Get<R::AccountId>>::get()
+	}
+}
+
+/// Deal with ethereum based priority fees/tips. See DealWithEthereumBaseFees for base fees.
+pub struct DealWithEthereumPriorityFees<R>(sp_std::marker::PhantomData<R>);
+impl<R> OnUnbalanced<Credit<R::AccountId, pallet_balances::Pallet<R>>>
+	for DealWithEthereumPriorityFees<R>
+where
+	R: pallet_balances::Config + pallet_author_inherent::Config,
+	pallet_author_inherent::Pallet<R>: Get<R::AccountId>,
+{
+	fn on_nonzero_unbalanced(amount: Credit<R::AccountId, pallet_balances::Pallet<R>>) {
+		ResolveTo::<BlockAuthorAccountId<R>, pallet_balances::Pallet<R>>::on_unbalanced(amount);
+	}
+}
```

Comment:
We could move this code and `DealWithSubstrateFeesAndTip` to common and share the implementation with all runtimes

Reply:
I kept it separate since the original `DealWithFees` was repeated in all runtimes with the same code